### PR TITLE
[vs17.8] Sync internal and public branches

### PR DESCRIPTION
**Pull request by YuliiaKovalova**
Fixes # https://github.com/dotnet/msbuild/issues/10824

### Context
Due to the gap between public and internal changes, mirroring doesn't work.

### Changes Made
Sync internal and public branches.

No functional changes, only infra.

### CODE REVIEW

File: .vsts-dotnet-ci.yml [hunk 1]
```diff
@@ -3,12 +3,28 @@ trigger:
...
+- name: NugetSecurityAnalysisWarningLevel
+  value: none
+
 jobs:
 - job: BootstrapMSBuildOnFullFrameworkWindows
   displayName: "Windows Full"
   pool:
     vmImage: 'windows-2022'
   steps:
+  - task: PowerShell@2
```
Comment (maridematte) [inline .vsts-dotnet-ci.yml:21]: We don't need these access to internal artifact feeds anymore. I think it is better if these are deleted from code.

File: eng/Packages.props [hunk 2]
```diff
@@ -28,7 +28,10 @@
     <PackageVersion Include="System.Runtime.CompilerServices.Unsafe" Version="$(SystemRuntimeCompilerServicesUnsafeVersion)" />
     <PackageVersion Include="System.Security.Principal.Windows" Version="$(SystemSecurityPrincipalWindowsVersion)" />
     <PackageVersion Include="System.Text.Encoding.CodePages" Version="$(SystemTextEncodingCodePagesVersion)" />
-    <PackageVersion Include="System.Text.Json" Version="$(SystemTextJsonVersion)" />
+    <!-- Ensure the latest STJ version for .NET-SDK delivered components -->
+    <PackageVersion Include="System.Text.Json" Version="$(SystemTextJsonVersion)" Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp'" />
+    <!-- Avoid bumping to STJ 8.0.4 for VS-delivered bits until VS updates later. This version was what was referenced in MSBuild 17.8.5 -->
```
Comment (maridematte) [inline eng/Packages.props:33]: I don't know if this still holds true too. @rainersigwald, can we just use the 8.0.4 of STJ for VS too?

File: src/MSBuild/MSBuild.csproj [hunk 3]
```diff
@@ -222,6 +222,8 @@
     <PackageReference Include="Microsoft.BuildXL.Processes" Condition="'$(FeatureReportFileAccesses)' == 'true'" PrivateAssets="all" />
     <PackageReference Include="Microsoft.IO.Redist" Condition="'$(FeatureMSIORedist)' == 'true'" />
     <PackageReference Include="System.Configuration.ConfigurationManager" />
+    <!-- Explicitly reference STJ in .NET 8 to bump to a patched version for the deps file -->
+    <PackageReference Include="System.Text.Json" Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' AND '$(DotNetBuildFromSource)' != 'true'" />
```
Comment (YuliiaKovalova) [inline src/MSBuild/MSBuild.csproj:226]: @rainersigwald is it still needed?

File: eng/Packages.props (hunk 2 above)
Comment (rainersigwald) [reply eng/Packages.props]: not for 17.8 AFAIK.

File: src/MSBuild/MSBuild.csproj (hunk 3 above)
Comment (rainersigwald) [reply src/MSBuild/MSBuild.csproj:226]: I think so.

**Review by maridematte — APPROVED** [review]

**Review by JanKrivanek — APPROVED** [review]

**Pull request merged**